arangod/Transaction/Context.h

```diff
@@ -141,6 +165,9 @@ class Context {
     TransactionId id;
     bool isReadOnlyTransaction;
     bool isFollowerTransaction;
+    bool isStreamingTransaction = false;
+    bool isEarlyReadOnly = false;
+    bool isJStransaction = false;
```

Comment:
I am not really happy with having separate booleans for `isStreamingTransaction` and `isJSTransaction`, because this allows to set both to `true` at the same time, which would at least in theory be invalid.

Reply:
I added assertions to be sure that both are not true 
```
 /// @brief sets the transaction to be streaming (used to know whether or not
  /// can read from query cache)
  void setStreaming() noexcept {
    TRI_ASSERT(_transaction.isJStransaction == false);
    _transaction.isStreamingTransaction = true;
  }

  /// @brief sets the transaction to be JS (used to know whether or not
  /// can read from query cache)
  void setJStransaction() noexcept {
    TRI_ASSERT(_transaction.isStreamingTransaction == false);
    _transaction.isJStransaction = true;
  }

```